modules/manage/pages/cluster-maintenance/nodewise-partition-recovery.adoc

```diff
@@ -0,0 +1,45 @@
+= Node-wise Partition Recovery
+:description: Feature to recover dead nodes.
+
+Multi-node or entire-AZ failures (especially in cloud environments) and some forms of human error may result in ‘stuck’ partitions with fewer replicas than required to make a quorum. In these scenarios, some data loss is unavoidable but admins need to recover what they can and move on.
+
+IMPORTANT: This is a last ditch measure when all other recovery options have failed. In some cases, no remaining replicas may exist for the partitions on the dead nodes. This recovery method is used in scenarios where you are already experiencing data loss - the goal is to stop the loss of any more data.
+
+== Prerequisites
+
+This feature has two prerequisites for recovering partitions:
+
+* Your cluster must have one or more `dead` nodes.
```

Comment:
@ztlpn is this really even true? The dead nodes could have been removed from the cluster long ago and there are still partitions without majority that need to be repaired.

We need to remove the emphasis on dead nodes here.

There are also NO pre-requisites to using this feature. you can use it when nothing is wrong at all. It will just skip healthy partitions.  It basically SCANS the nodes for anything without majority and repairs these partitions. That's it.

Lets drop this 'prerequisite' section entirely, its not needed and a bit misleading

Reply:
Summoning @bharathv as the SME

Reply:
Oops sorry @ztlpn

Reply:
> is this really even true? The dead nodes could have been removed from the cluster long ago and there are still partitions without majority that need to be repaired.

Technically that is not possible because decommissioning a dead node requires draining all replicas on it and a partition with lost majority cannot do it. That said, I agree that we need remove too much emphasis on the dead nodes here (see below).

> There are also NO pre-requisites to using this feature. you can use it when nothing is wrong at all. It will just skip healthy partitions. It basically SCANS the nodes for anything without majority and repairs these partitions. That's it.

@mattschumpert  A small clarification, when it scans the topic list, the decision of majority loss is based on the user input nodes. Example, partition p has replicas [0, 1, 2] and user says "unsafe recover 1,2", Redpanda knows that partition `p` lost majority because majority [1, 2] of its current replica assignment [0,1,2] are gone. In this process, the logic actually is based on the user input to recover partitions from nodes [1,2]. Without that the broker has no way of knowing whether nodes 1, 2 are dead or down temporarily or partitioned from the cluster etc.

So, I wouldn't say there are NO prereqs. The prereq here is that the nodes are irrecoverable (data is gone for whatever reason, dead, corruption, vanished etc) and the user has confirmed that with their tooling and the user has to communicate that to Redpanda with this command.


The way I'd phrase it is, if there is a majority loss of partitions because one or more nodes are no longer accessible and the data is irrecoverable, we can use this command to unsafely recover them using whatever replicas are remaining. The reason I think `dead nodes` is confusing is because there is no technical definition of what `dead` is. May be it is dead because someone shut it down but the data is recoverable once someone reboots it. So I think we should rather emphasis on "non recoverability" of data from those nodes.

wdyt.

Reply:
I reworded this as an assumptions section, per this feedback and the feedback below.